### PR TITLE
Fix swatches for new product widget

### PR DESCRIPTION
**Pull request by sunnykhatri**
### Fixed Issues (if relevant)

1. magento/magento2#14134: Swatches not showing on homepage New products widget

### Manual testing scenarios (*)

1. Create a new **Catalog New Products List** widget and call that widget on HomPage
2. Add any of configurable product in widget

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=22661) <br/>All committers have signed the CLA.

**Comment by sunnykhatri** [comment]
@magento-engcom-team Any update on this request?
Thanks

File: app/code/Magento/Catalog/Block/Product/Widget/NewWidget.php [hunk 1]
```diff
@@ -65,6 +86,8 @@ public function __construct(
         \Magento\Catalog\Model\Product\Visibility $catalogProductVisibility,
         \Magento\Framework\App\Http\Context $httpContext,
         array $data = [],
+        LayoutFactory $layoutFactory = null,
+        EncoderInterface $urlEncoder = null,
```
Comment (sidolov) [inline app/code/Magento/Catalog/Block/Product/Widget/NewWidget.php:90]: Please, add new dependencies at the end of the method.

**Review by sidolov — CHANGES_REQUESTED** [review]
Could you please clarify a few moments regarding the provided solution: 
- Why we need new layout handle instead of using existing renderers listing? 
- Can we reuse existing phtml templates to achive the same behavior?

**Comment by sidolov** [comment]
@sunnykhatri , I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

**Pull request closed without merging**